doc/mgmt/Management Framework.md

```diff
@@ -0,0 +1,2029 @@
+
+#### 3.1.2 Run time flow
+
+##### 3.1.2.1 CLI
+
+1. CLI uses the KLISH framework to provide a CLI shell. The CLI request is converted to a corresponding REST client SDK request that was generated by the Swagger generator, and is given to the REST server.
+2. The Swagger generated REST server handles all the REST requests from the client SDK and invokes a common handler for all the create, update, replace, delete and get operations along with path and payload. This common handler converts all the requests into Translib arguments and invokes the corresponding Translib provided APIs.
+3. Translib uses the value of the input (incoming) path/URI to determine the identity of the appropriate App module. It then calls that App module, passing the request as either a populated YGOT structure or as JSON, depending upon the data modelling method in use for the application (see section [3.1.1](#311-build-time-flow)).
+4. Further processing of CLI commands will be handled by Translib componenets that will be discussed in more detail in the later sections.
+
+##### 3.1.2.2 REST
+
+1. REST client will use the Swagger generated client SDK to send the request to the REST server.
+2. From then on the flow is similar to the one seen in the CLI.
+
+##### 3.1.2.3 gNMI
+
+GNMI service defines a gRPC-based protocol for the modification and retrieval of configuration from a target device, as well as the control and generation of telemetry streams from a target device to a data collection system. Refer  [GNMI spec](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md)
+
+![GNMI Service High level diagram (Proposed)](images/GNMI_Server.png)
+
+1. Existing SONiC telemetry framework has been extended to support the new GNMI services.
+2. All 4 GNMI services are supported: Get, Set, Capabilities and Subscribe.
+3. A new transl data client is added to process the incoming YANG-based request (either standard or proprietary)
+4. The new transl data client relies on Translib infra provided to translate, get, set of the YANG objects.
+
+More details onthe GNMI server, Client and workflow provided later in the document.
+
+### 3.2 SONiC Management Framework Components
+
+Management framework components can be classified into
+
+1. Build time components
+2. Run time components
+
+#### 3.2.1 Build time components
+
+Following are the build time components of the management framework
+
+1. Pyang compiler (for YANG to OpenAPI conversion)
+2. Swagger generator
+3. YGOT generator
+4. pyang compiler (for YANG to YIN conversion)
```

Comment:
Are these components compatible with Automake and/or do they support an incremental build approach? More generally, if I add a new object or tweak the definition of an object slightly does it require a complete re-build of these components or just the dependencies?

Reply:
All code generators and compilers are integrated with SONiC make infrastructure. Incremental build is supported.

Reply:
In general the components are having individual Makefiles manually written where dependencies are captured and they support incremental build approach.